akka-bbb-apps/src/main/scala/org/bigbluebutton/core/models/Plugins.scala

```diff
@@ -96,17 +98,56 @@ object PluginModel {
     val pluginWithAbsoluteJsEntrypoint = replaceRelativeJavascriptEntrypoint(plugin)
     replaceRelativeLocalesBaseUrl(pluginWithAbsoluteJsEntrypoint)
   }
-  def createPluginModelFromJson(json: util.Map[String, AnyRef]): PluginModel = {
+
+  private def checkPluginSdkVersionDevMode(pluginSdkVersion: String): Boolean =
+    pluginSdkVersion.startsWith("https://codeload")
+
+  private def html5SdkSatisfiesPluginRequiredVersion(bbbHtml5SdkVersion: String, pluginHtml5SdkRequirement: String): Boolean = {
+    try {
+      val v = Version.parse(bbbHtml5SdkVersion)
+      v.satisfies(pluginHtml5SdkRequirement)
+    } catch {
+      case e: Exception =>
+        logger.error(s"Unexpected error while parsing SDK versions: $e")
+        false
+    }
+  }
+
+  private def isServerSdkCompatibleWithPlugin(html5PluginSdkVersion: String, pluginManifestRequiredSdkVersion: String): Boolean = {
+    // Returns `true` if:
+    // - Both versions are URLs starting with "https://codeload" (development mode).
+    // - OR bbb-html5 SDK version satisfies the manifest's required SDK version.
+    (checkPluginSdkVersionDevMode(html5PluginSdkVersion) && checkPluginSdkVersionDevMode(pluginManifestRequiredSdkVersion)) ||
```

Comment:
I'm wondering why the developer needs to set `https://codeload` in both `bigbluebutton.properties` and `manifest.json`.

To me, it makes more sense to use `https://codeload` only in `package.json`, and specify the next SDK version number in `bigbluebutton.properties` and `manifest.json` (if needed).

Having to update multiple config files feels a bit excessive. Updating just `package.json` seems more straightforward.
What do you think?

If you agree this validation should be removed.

Reply:
Agreed, it makes sense to reduce the number of places we mention `codeload` in our code.